op-supervisor/supervisor/backend/syncnode/node.go

```diff
@@ -420,6 +434,34 @@ func (m *ManagedNode) Close() error {
 	return nil
 }
 
+// replaceIfNeeded checks the node's last local safe block against the backend
+// to see if it should be replaced. If so, it invalidates the block and returns true.
+// if replacement doesn't happen, it returns false.
+// This codepath is triggered when a node is syncing,
+// and the Supervisor has already dealth with the invalidation and replacement.
+// nodes which are catching up need to know if a replacement is canonical at this point,
+func (m *ManagedNode) replaceIfNeeded() bool {
+	ctx, cancel := context.WithTimeout(m.ctx, internalTimeout)
+	defer cancel()
+	err := m.backend.IsReplacement(ctx, m.chainID, m.lastNodeLocalSafe)
+	if err != nil {
+		m.log.Info("Replacement block detected, initiating replacement",
+			"lastNodeLocalSafe", m.lastNodeLocalSafe,
+			"err", err)
```

Comment:
Why `err != nil`? Wouldn't it also return an error if the block in the DB is an invalidated block?
I think it returns an error only if the block is *not* a replacement, the opposite of what we're trying to do here.
Can you add an action test to show that a new node can sync, after a prior replacement?

Reply:
Ah yeah you're right, I'm returning `nil` if it *is* a replacement, and then I flipped my logic here.